tests/Scheduler/DbSchedulerTest.php

```diff
@@ -131,4 +133,169 @@ public function testRemove(): void
         $scheduler = new DbScheduler($this->adapter);
         static::assertTrue($scheduler->remove(234));
     }
+
+    public function testRetrieveBatchWithNoResults(): void
+    {
+        $stmt = $this->createMock(\PDOStatement::class);
+        $stmt->expects(static::once())
+            ->method('rowCount')
+            ->willReturn(0);
+        $this->adapter->expects(static::once())
+            ->method('query')
+            ->willReturn($stmt);
+        $scheduler = new DbScheduler($this->adapter);
+        static::assertFalse($scheduler->retrieveBatch());
+    }
+
+    public function testRetrieveBatchWithResults(): void
+    {
+        $rowData = [
+            [
+                'id' => 1,
+                'tube' => 'queue1',
+                'data' => serialize('data1'),
+                'scheduled_ts' => date('Y-m-d H:i:s'),
+                'priority' => 1024,
+                'ttr' => 60,
+            ],
+            [
+                'id' => 2,
+                'tube' => 'queue2',
+                'data' => serialize('data2'),
+                'scheduled_ts' => date('Y-m-d H:i:s'),
+                'priority' => 512,
+                'ttr' => 30,
+            ],
+        ];
+
+        $updateStmt = $this->createMock(\PDOStatement::class);
+        $updateStmt->expects(static::once())
+            ->method('rowCount')
+            ->willReturn(2);
+
+        $selectStmt = $this->createMock(\PDOStatement::class);
+        $selectStmt->expects(static::once())
+            ->method('fetchAll')
+            ->willReturn($rowData);
+
+        $this->adapter->expects(static::exactly(2))
+            ->method('query')
+            ->willReturnOnConsecutiveCalls($updateStmt, $selectStmt);
+
+        $scheduler = new DbScheduler($this->adapter);
+        $jobs = $scheduler->retrieveBatch();
+
+        static::assertIsArray($jobs);
+        static::assertCount(2, $jobs);
+        static::assertEquals(1, $jobs[0]['id']);
```

Comment:
Suggest using `assertSame()` instead of `assertEquals()`, which is the equivalent of using `===` instead of `==`.